Updated and fixed ZMX Metric Reference docs.

Grryum opened the pull request:
Made docs up to date according to latest ZIO Metric API.

partially closes #21

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/zio/zio-metrics-connectors?pullRequest=43) <br/>All committers have signed the CLA.

petoalbert commented on docs/metrics/zmx-metric-reference.md:
```diff
@@ -8,72 +8,79 @@ import zio._
+All metrics in ZMX are defined in the form of aspects that can be applied to effects without changing
+the signature of the effect it is applied to.
+
+Also, every `Metric`s implementation (kind of metric) are further qualified by a type parameter `In` that must be
```
What did you mean by "(kind of metric)" here? Can we remove it?

petoalbert commented on docs/metrics/zmx-metric-reference.md:
```diff
@@ -8,72 +8,79 @@ import zio._
+metric, the API defines a `contramap` method to construct a `Metric[_, In2, _]` with a mapper function
+from `In` to the type required by the metric.
+
+There is also an ability to setting up additional conditions for metric value capture.
```
I believe the correct grammatical way would be to write "to set up" instead of "to setting up".

petoalbert commented on docs/metrics/zmx-metric-reference.md:
````diff
@@ -97,102 +104,70 @@ val countBytes = nextDoubleBetween(0.0d, 100.0d) @@ aspCountBytes
 ```
 
-Now we can apply these effects to effects having an output type `Double`. Note that we can instrument 
+Now we can apply these effects to effects having an output type `Double`. Note that we can instrument
````
I think it should be "apply these aspects to effects".

petoalbert reviewed (changes requested):
Thank you @Grryum for these changes. I left a few minor comments, otherwise everything looks good. Could you please sign the CLA as well?

Grryum committed:
Fixed some typo and removed any ZMX reference from metric reference doc.

Grryum committed:
Updated and fixed ZMX Metric Reference docs.

Grryum commented:
@petoalbert thank you for your review! I fixed typos and removed any ZMX reference from this part of doc as we discussed with @adamgfraser in Discord.
I suppose that ZMX references should also be removed from other parts of doc.
I could do it in this PR or in another one.

petoalbert commented:
Thank you @Grryum . If you could rename all the other ZMX references in this PR, that would be nice.

Grryum committed:
Removed ZMX references from index.md doc.

Grryum committed:
Removed ZMX reference from docs and sources.

Grryum commented:
@petoalbert I tried to remove all ZMX references from docs and sources. Also tried to actualize samples.
I did not touch main README.md and could not recreate some of screenshots with ZMX references. Don't know if it matters.
Please check it out.
@calvinlfer your turn :)

petoalbert commented:
Thank you @Grryum! I think the README needs a significant update anyway, so we can do that later.
